### PR TITLE
#4008 - Allow packages and payloads over 2GiB

### PR DESCRIPTION
**Pull request by AlexKubiesa**
This should resolve wixtoolset/issues/issues/4008.

The fix itself was simple - just converting an int to a long.

For the test, I basically copied the existing pattern in `BundleFixture`. The data file `large_file.dat` committed to the repo is a dummy file which gets enlarged to 2.5GiB during the test. It then gets deleted at the end of the test to avoid taking up a lot of space.

### CODE REVIEW

File: src/test/WixToolsetTest.CoreIntegration/WixToolsetTest.CoreIntegration.csproj [hunk 1]
```diff
@@ -12,6 +12,18 @@
     <Content Include="TestData\**" CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Remove="TestData\LargePayload\Bundle.en-us.wxl" />
```
Comment (robmen) [inline src/test/WixToolsetTest.CoreIntegration/WixToolsetTest.CoreIntegration.csproj:16]: This block is an artifact of adding files via VS, you don't need it.

File: src/test/WixToolsetTest.CoreIntegration/BundleFixture.cs [hunk 2]
```diff
@@ -366,5 +366,103 @@ public void CantBuildWithUnscheduledRollbackBoundary()
...
+                Assert.True(File.Exists(pdbPath));
+
+                using (var wixOutput = WixOutput.Read(pdbPath))
+                {
+
+                    var intermediate = Intermediate.Load(wixOutput);
+                    var section = intermediate.Sections.Single();
+
+                    var bundleSymbol = section.Symbols.OfType<WixBundleSymbol>().Single();
+                    Assert.Equal("1.0.0.0", bundleSymbol.Version);
```
Comment (robmen) [inline src/test/WixToolsetTest.CoreIntegration/BundleFixture.cs:417]: Do we need to validate all these pieces of information? Most of it is covered by other targeted tests. It'd be best if this one focuses on the large file facets.

File: src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/large_file.dat [hunk 3]
```diff
@@ -0,0 +1,2 @@
+When running the tests, this file will be overwritten with 2.5GB of data to test how Wix handles large files. We've avoided
```
Comment (robmen) [inline src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/large_file.dat:1]: But this file will always end up "edited" in git after a test run, correct? So, would be easy for someone (else) to accidentally commit the file. Probably better not to put the file in the repository or make sure it's ignored.

**Review by robmen — CHANGES_REQUESTED** [review]

File: src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/large_file.dat (hunk 3 above)
Comment (AlexKubiesa) [reply src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/large_file.dat:1]: The file is only edited in the build output directory. The source file is left alone.

**Comment by robmen** [comment]
One last nit: could you rebase these changes down to a single commit? I prefer to see the change and its test(s) all together in the same commit for future reference. Also, the separate commits for review feedback add noise to the history. Thanks.

PS: Sorry, should have mentioned that in last review.

**Comment by AlexKubiesa** [comment]
Ok no worries, I'll do that now.

**Review by robmen — APPROVED** [review]

File: src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/fakeba.dll [hunk 4]
```diff
@@ -0,0 +1 @@
+This is a fakeba.dll
```
Comment (rseanhall) [inline src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/fakeba.dll:1]: @robmen Like you wanted them to narrow what was being tested, I would like tests not to have their own files unless it's necessary. Creating fake BA dlls and using localization seems like overkill in most scenarios and will also bloat the repo.

Comment (robmen) [reply src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/fakeba.dll:1]: I don't disagree but we should clean up all the tests to work this way.

Comment (rseanhall) [reply src/test/WixToolsetTest.CoreIntegration/TestData/LargePayload/data/fakeba.dll:1]: I already write all of my tests that way, I haven't looked lately at how many need to be cleaned up.

**Pull request merged**